frontend/src/DatasetList.js

```diff
@@ -1,119 +1,198 @@
       });
-    }).catch(err => {
-      if (err.status && err.status === 401) {
-        // Server returned 401 which means token was revoked
-        document.cookie = 'token=;max-age=0;path=/';
-        window.location.reload();
-      }
-      this.setState({
-        error: err
+    return () => {
+      getNewData.abort();
+    };
+  }, []);
+
+  useEffect(() => {
+    getData
+      .set("Authorization", "Bearer " + props.app_token)
+      .then((req) => {
+        setDatasets(req.body);
+      })
+      .catch((err) => {
+        if (err.status && err.status === 401) {
+          // Server returned 401 which means token was revoked
+          document.cookie = "token=;max-age=0;path=/";
+          window.location.reload();
+        }
+        setError(err);
+        //setToastMsg({msg: error.response.text, lvl: "danger"});
       });
-    });
-  }
 
-  componentWillUnmount() {
-    this.getData.abort();
-    this.startImport.abort();
-  }
+    return () => {
+      getData.abort();
+    };
+  }, [newDatasets]);
 
-  import = () => {
-    return this.startImport.set('Authorization', 'Bearer ' + this.props.app_token).then(() => {
-      window.location.reload();
-    }).catch(err => {
-      if (err.status && err.status === 401) {
-        // Server returned 401 which means token was revoked
-        document.cookie = 'token=;max-age=0;path=/';
-        window.location.reload();
-      }
-      this.setState({
-        error: err
-      });
-    });
-  }
+  useEffect(() => {
+    setToastMsg(
+      !Array.isArray(error)
+        ? {msg: error.response.text, lvl: "danger"}
+        : {msg: ""}
+    );
+  }, [error]);
 
-  render() {
-    const datasets = this.state.datasets.map(dataset => {
-      return (
-        <tr key={dataset.id}>
-          <td>{dataset.name}</td>
-          <td>{dataset.type}</td>
-          <td>{dataset.files_type}</td>
-          <td>{dataset.files_count}</td>
-          <td>{new Date(dataset.start_date).toDateString()}</td>
-          <td>{new Date(dataset.end_date).toDateString()}</td></tr>
-      );
-    });
+  const importNewData = (wanted_datasets) => {
+    return startImport
+      .set("Authorization", "Bearer " + props.app_token)
+      .send({wanted_datasets})
+      .set("Accept", "application/json")
+      .then((req) => {
+        const remainingDatasets = newDatasets.filter((newDataset) =>
+          req.body.some(
+            (importedDataset) => importedDataset.name !== newDataset.name
+          )
+        );
+        setNewDatasets(remainingDatasets);
+        setError([]);
+        setToastMsg([]);
+        setOpenModal(false);
+      })
+      .catch((err) => {
+        if (err.status && err.status === 401) {
+          // Server returned 401 which means token was revoked
+          document.cookie = "token=;max-age=0;path=/";
+          window.location.reload();
+        }
+        setError(err);
+        setOpenModal(false);
+      });
+  };
```

Comment:
As seen with @Qouagga: the body of this function should be in a `useEffect` hook as it's an asynchronous call to the API.
Best way to do it is to create a boolean dedicated `isImportReady` or `launchImport`, pass this boolean to the modal instead of the `importNewData` function, and use it as a followed value by the useEffet hook.
The hook checks the value of the boolean, sets it to `false` immediatly and launches the API.

Reply:
And don't forget to `.abort()` ^^